### ISSUE TITLE
How to augment my own dataset using the additionnal annotations, i.e., how to generate the Aug files

### ISSUE THREAD

**Issue by JennyVVV** [yassouali/pytorch_segmentation#28]
Dear Author,

I have labeled my own datasets, now I have RGB images and the label png data, how can I generate the SegmentationClassAug files as the VOC datasets? Hope for your quick reply, thank you

**Comment by yassouali** [comment]
Hi, no need to create SegmentationClassAug, that just for pascal voc, you can organize your files however you like. All you need to do is create you dataloader / dataset, like the ones in `dataloaders/`. 

For the dataset, you'll need a method `_set_files` where you set the filenames / paths of  your images and your labels, and `_load_data` where you'll load a given image and labels given the passed `id`.

Take a look at the examples in  `dataloaders/`, you'll need to do the same thing.

**Comment by JennyVVV** [comment]
> Hi, do you need to train on your own data ? no need to create SegmentationClassAug, that just for pascal voc, you can organize your files however you like. All you need to do is create you dataloader / dataset, like the ones in `dataloaders/`.
> For the dataset, you'll need a method `_set_files` where you set the filenames / paths of your images and your labels, and `_load_data` where you'll load a given image and labels given the passed `id`.
> Take a look at the examples in `dataloaders/`, you'll need to do the same thing.
> 
> If your asking how to generate augmented labels, the augmented set of pascal voc was manual annotated, you can find the description here (http://home.bharathh.info/pubs/pdfs/BharathICCV2011.pdf), if you want to automatically label you unlabeled data, see pseudo labeling or some other semi supervised techniques.

Hi, thank you for your quick reply. I think I can rewrite the dataloaders for my own datasets. I have one more question.  Because I labeled my datasets using the "labelme" program, so the annotations are all color PNG images with different colors according to different classes. When I inference, do I need to prepare the palette by myself (like ADE20K_palette, CityScpates_palette, COCO_palette) or just using the get_voc_palette(num_classes) function? How can I guarantee the results of testing images have consistent colors with the training color labels?

**Comment by yassouali** [comment]
The palette is used just for displaying the prediction, it has nothing to do with training, in training you'll use the labels (0 - C-1, with C as the number of classes), so in you dataloader, the labels mush be of shape H x W, where each element is the class label. You can then use any palette you want, it is not really important.
